<?xml version='1.0' encoding='utf-8'?>
<sdf version="1.5">
  <world name="default">
    <include>
      <uri>model://sun</uri>
    </include>
    <include>
      <uri>model://ground_plane</uri>
      <pose>0 0 -0.001 0 0 0</pose>
    </include>
    <model name="airspace">
      <pose frame="">0 0 2.5 0 0 0</pose>
      <link name="link_0">
        <pose frame="">0 0 0 0 0 0</pose>
        <gravity>0</gravity>
        <self_collide>0</self_collide>
        <kinematic>1</kinematic>
        <visual name="airspace_visual">
          <pose frame="">0 0 0 0 0 0</pose>
          <geometry>
            <box>
              <size>20 20 5.0</size>
            </box>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/DarkOrangeTransparentOverlay</name>
            </script>
          </material>
          <cast_shadows>0</cast_shadows>
          <transparency>0.85</transparency>
        </visual>
      </link>
    </model>
    <model name="aruco_marker_1">
      <pose frame="''">0.5 -0.5 0.0 0.0 0.0 0.785398163397</pose>
      <link name="link_0">
        <pose frame="''">0 0 0 0 0 0</pose>
        <visual name="visual">
          <geometry>
            <box>
              <size>0.197 0.197 1e-5</size>
            </box>
          </geometry>
          <material>
            <script>
              <uri>file:///home/dduberg/catkin_ws/src/dd2419/dd2419_simulation/markers/0.197x0.197/scripts</uri>
              <uri>file:///home/dduberg/catkin_ws/src/dd2419/dd2419_simulation/markers/0.197x0.197/textures</uri>
              <name>marker_aruco-1/marker</name>
            </script>
          </material>
          <pose frame="''">0 0 0.001 0 0 0</pose>
          <cast_shadows>1</cast_shadows>
          <transparency>0</transparency>
        </visual>
        <collision name="collision">
          <pose frame="''">0 0 0.001 0 0 0</pose>
          <geometry>
            <box>
              <size>0.197 0.197 1e-5</size>
            </box>
          </geometry>
        </collision>
        <visual name="visual_backside">
          <pose frame="''">0 0 0 0 0 0</pose>
          <geometry>
            <box>
              <size>0.197 0.197 1e-5</size>
            </box>
          </geometry>
          <material>
            <script>Gazebo/Grey</script>
          </material>
        </visual>
        <collision name="collision_backside">
          <pose frame="''">0 0 0 0 0 0</pose>
          <geometry>
            <box>
              <size>0.197 0.197 1e-5</size>
            </box>
          </geometry>
        </collision>
      </link>
      <static>1</static>
      <allow_auto_disable>1</allow_auto_disable>
    </model>
    <model name="aruco_marker_2">
      <pose frame="''">0.5 0.5 0.0 0.0 0.0 2.35619449019</pose>
      <link name="link_0">
        <pose frame="''">0 0 0 0 0 0</pose>
        <visual name="visual">
          <geometry>
            <box>
              <size>0.197 0.197 1e-5</size>
            </box>
          </geometry>
          <material>
            <script>
              <uri>file:///home/dduberg/catkin_ws/src/dd2419/dd2419_simulation/markers/0.197x0.197/scripts</uri>
              <uri>file:///home/dduberg/catkin_ws/src/dd2419/dd2419_simulation/markers/0.197x0.197/textures</uri>
              <name>marker_aruco-2/marker</name>
            </script>
          </material>
          <pose frame="''">0 0 0.001 0 0 0</pose>
          <cast_shadows>1</cast_shadows>
          <transparency>0</transparency>
        </visual>
        <collision name="collision">
          <pose frame="''">0 0 0.001 0 0 0</pose>
          <geometry>
            <box>
              <size>0.197 0.197 1e-5</size>
            </box>
          </geometry>
        </collision>
        <visual name="visual_backside">
          <pose frame="''">0 0 0 0 0 0</pose>
          <geometry>
            <box>
              <size>0.197 0.197 1e-5</size>
            </box>
          </geometry>
          <material>
            <script>Gazebo/Grey</script>
          </material>
        </visual>
        <collision name="collision_backside">
          <pose frame="''">0 0 0 0 0 0</pose>
          <geometry>
            <box>
              <size>0.197 0.197 1e-5</size>
            </box>
          </geometry>
        </collision>
      </link>
      <static>1</static>
      <allow_auto_disable>1</allow_auto_disable>
    </model>
    <model name="aruco_marker_3">
      <pose frame="''">-0.5 0.5 0.0 0.0 0.0 -2.35619449019</pose>
      <link name="link_0">
        <pose frame="''">0 0 0 0 0 0</pose>
        <visual name="visual">
          <geometry>
            <box>
              <size>0.197 0.197 1e-5</size>
            </box>
          </geometry>
          <material>
            <script>
              <uri>file:///home/dduberg/catkin_ws/src/dd2419/dd2419_simulation/markers/0.197x0.197/scripts</uri>
              <uri>file:///home/dduberg/catkin_ws/src/dd2419/dd2419_simulation/markers/0.197x0.197/textures</uri>
              <name>marker_aruco-3/marker</name>
            </script>
          </material>
          <pose frame="''">0 0 0.001 0 0 0</pose>
          <cast_shadows>1</cast_shadows>
          <transparency>0</transparency>
        </visual>
        <collision name="collision">
          <pose frame="''">0 0 0.001 0 0 0</pose>
          <geometry>
            <box>
              <size>0.197 0.197 1e-5</size>
            </box>
          </geometry>
        </collision>
        <visual name="visual_backside">
          <pose frame="''">0 0 0 0 0 0</pose>
          <geometry>
            <box>
              <size>0.197 0.197 1e-5</size>
            </box>
          </geometry>
          <material>
            <script>Gazebo/Grey</script>
          </material>
        </visual>
        <collision name="collision_backside">
          <pose frame="''">0 0 0 0 0 0</pose>
          <geometry>
            <box>
              <size>0.197 0.197 1e-5</size>
            </box>
          </geometry>
        </collision>
      </link>
      <static>1</static>
      <allow_auto_disable>1</allow_auto_disable>
    </model>
    <model name="aruco_marker_4">
      <pose frame="''">-0.5 -0.5 0.0 0.0 0.0 -0.785398163397</pose>
      <link name="link_0">
        <pose frame="''">0 0 0 0 0 0</pose>
        <visual name="visual">
          <geometry>
            <box>
              <size>0.197 0.197 1e-5</size>
            </box>
          </geometry>
          <material>
            <script>
              <uri>file:///home/dduberg/catkin_ws/src/dd2419/dd2419_simulation/markers/0.197x0.197/scripts</uri>
              <uri>file:///home/dduberg/catkin_ws/src/dd2419/dd2419_simulation/markers/0.197x0.197/textures</uri>
              <name>marker_aruco-4/marker</name>
            </script>
          </material>
          <pose frame="''">0 0 0.001 0 0 0</pose>
          <cast_shadows>1</cast_shadows>
          <transparency>0</transparency>
        </visual>
        <collision name="collision">
          <pose frame="''">0 0 0.001 0 0 0</pose>
          <geometry>
            <box>
              <size>0.197 0.197 1e-5</size>
            </box>
          </geometry>
        </collision>
        <visual name="visual_backside">
          <pose frame="''">0 0 0 0 0 0</pose>
          <geometry>
            <box>
              <size>0.197 0.197 1e-5</size>
            </box>
          </geometry>
          <material>
            <script>Gazebo/Grey</script>
          </material>
        </visual>
        <collision name="collision_backside">
          <pose frame="''">0 0 0 0 0 0</pose>
          <geometry>
            <box>
              <size>0.197 0.197 1e-5</size>
            </box>
          </geometry>
        </collision>
      </link>
      <static>1</static>
      <allow_auto_disable>1</allow_auto_disable>
    </model>
    <model name="wall_0">
      <pose frame="''">0 0 0 0 0 0</pose>
      <link name="link_0">
        <pose frame="">0 0 0 0 0 0</pose>
        <gravity>0</gravity>
        <self_collide>0</self_collide>
        <kinematic>1</kinematic>
        <visual name="wall_0_visual_0">
          <pose frame="">2.0 3.0 1.5 0 0 1.57079632679</pose>
          <geometry>
            <box>
              <size>2.0 0.02 3.0</size>
            </box>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/DarkOrangeTransparentOverlay</name>
            </script>
          </material>
          <cast_shadows>1</cast_shadows>
          <transparency>0</transparency>
        </visual>
        <collision name="wall_0_collision_0">
          <pose frame="">2.0 3.0 1.5 0 0 1.57079632679</pose>
          <geometry>
            <box>
              <size>2.0 0.02 3.0</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>
    <model name="gate_1">
      <pose frame="''">0 0 0 0 0 0</pose>
      <link name="link_0">
        <pose frame="">0 0 0 0 0 0</pose>
        <gravity>0</gravity>
        <self_collide>0</self_collide>
        <kinematic>1</kinematic>
        <visual name="gate_1_visual_0">
          <pose frame="">0.323223304703 -0.323223304703 0.2 0 0 -0.785398163397</pose>
          <geometry>
            <box>
              <size>0.1 0.02 0.4</size>
            </box>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/DarkOrangeTransparentOverlay</name>
            </script>
          </material>
          <cast_shadows>1</cast_shadows>
          <transparency>0</transparency>
        </visual>
        <collision name="gate_1_collision_0">
          <pose frame="">0.323223304703 -0.323223304703 0.2 0 0 -0.785398163397</pose>
          <geometry>
            <box>
              <size>0.1 0.02 0.4</size>
            </box>
          </geometry>
        </collision>
        <pose frame="">0 0 0 0 0 0</pose>
        <gravity>0</gravity>
        <self_collide>0</self_collide>
        <kinematic>1</kinematic>
        <visual name="gate_1_visual_1">
          <pose frame="">0.676776695297 -0.676776695297 0.2 0 0 -0.785398163397</pose>
          <geometry>
            <box>
              <size>0.1 0.02 0.4</size>
            </box>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/DarkOrangeTransparentOverlay</name>
            </script>
          </material>
          <cast_shadows>1</cast_shadows>
          <transparency>0</transparency>
        </visual>
        <collision name="gate_1_collision_1">
          <pose frame="">0.676776695297 -0.676776695297 0.2 0 0 -0.785398163397</pose>
          <geometry>
            <box>
              <size>0.1 0.02 0.4</size>
            </box>
          </geometry>
        </collision>
        <pose frame="">0 0 0 0 0 0</pose>
        <gravity>0</gravity>
        <self_collide>0</self_collide>
        <kinematic>1</kinematic>
        <visual name="gate_1_visual_2">
          <pose frame="">0.5 -0.5 1.2 0 0 -0.785398163397</pose>
          <geometry>
            <box>
              <size>0.6 0.02 1.6</size>
            </box>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/DarkOrangeTransparentOverlay</name>
            </script>
          </material>
          <cast_shadows>1</cast_shadows>
          <transparency>0</transparency>
        </visual>
        <collision name="gate_1_collision_2">
          <pose frame="">0.5 -0.5 1.2 0 0 -0.785398163397</pose>
          <geometry>
            <box>
              <size>0.6 0.02 1.6</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>
    <model name="gate_2">
      <pose frame="''">0 0 0 0 0 0</pose>
      <link name="link_0">
        <pose frame="">0 0 0 0 0 0</pose>
        <gravity>0</gravity>
        <self_collide>0</self_collide>
        <kinematic>1</kinematic>
        <visual name="gate_2_visual_0">
          <pose frame="">0.676776695297 0.676776695297 0.2 0 0 -2.35619449019</pose>
          <geometry>
            <box>
              <size>0.1 0.02 0.4</size>
            </box>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/DarkOrangeTransparentOverlay</name>
            </script>
          </material>
          <cast_shadows>1</cast_shadows>
          <transparency>0</transparency>
        </visual>
        <collision name="gate_2_collision_0">
          <pose frame="">0.676776695297 0.676776695297 0.2 0 0 -2.35619449019</pose>
          <geometry>
            <box>
              <size>0.1 0.02 0.4</size>
            </box>
          </geometry>
        </collision>
        <pose frame="">0 0 0 0 0 0</pose>
        <gravity>0</gravity>
        <self_collide>0</self_collide>
        <kinematic>1</kinematic>
        <visual name="gate_2_visual_1">
          <pose frame="">0.323223304703 0.323223304703 0.2 0 0 -2.35619449019</pose>
          <geometry>
            <box>
              <size>0.1 0.02 0.4</size>
            </box>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/DarkOrangeTransparentOverlay</name>
            </script>
          </material>
          <cast_shadows>1</cast_shadows>
          <transparency>0</transparency>
        </visual>
        <collision name="gate_2_collision_1">
          <pose frame="">0.323223304703 0.323223304703 0.2 0 0 -2.35619449019</pose>
          <geometry>
            <box>
              <size>0.1 0.02 0.4</size>
            </box>
          </geometry>
        </collision>
        <pose frame="">0 0 0 0 0 0</pose>
        <gravity>0</gravity>
        <self_collide>0</self_collide>
        <kinematic>1</kinematic>
        <visual name="gate_2_visual_2">
          <pose frame="">0.5 0.5 1.2 0 0 -2.35619449019</pose>
          <geometry>
            <box>
              <size>0.6 0.02 1.6</size>
            </box>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/DarkOrangeTransparentOverlay</name>
            </script>
          </material>
          <cast_shadows>1</cast_shadows>
          <transparency>0</transparency>
        </visual>
        <collision name="gate_2_collision_2">
          <pose frame="">0.5 0.5 1.2 0 0 -2.35619449019</pose>
          <geometry>
            <box>
              <size>0.6 0.02 1.6</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>
    <model name="gate_3">
      <pose frame="''">0 0 0 0 0 0</pose>
      <link name="link_0">
        <pose frame="">0 0 0 0 0 0</pose>
        <gravity>0</gravity>
        <self_collide>0</self_collide>
        <kinematic>1</kinematic>
        <visual name="gate_3_visual_0">
          <pose frame="">-0.323223304703 0.323223304703 0.2 0 0 2.35619449019</pose>
          <geometry>
            <box>
              <size>0.1 0.02 0.4</size>
            </box>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/DarkOrangeTransparentOverlay</name>
            </script>
          </material>
          <cast_shadows>1</cast_shadows>
          <transparency>0</transparency>
        </visual>
        <collision name="gate_3_collision_0">
          <pose frame="">-0.323223304703 0.323223304703 0.2 0 0 2.35619449019</pose>
          <geometry>
            <box>
              <size>0.1 0.02 0.4</size>
            </box>
          </geometry>
        </collision>
        <pose frame="">0 0 0 0 0 0</pose>
        <gravity>0</gravity>
        <self_collide>0</self_collide>
        <kinematic>1</kinematic>
        <visual name="gate_3_visual_1">
          <pose frame="">-0.676776695297 0.676776695297 0.2 0 0 2.35619449019</pose>
          <geometry>
            <box>
              <size>0.1 0.02 0.4</size>
            </box>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/DarkOrangeTransparentOverlay</name>
            </script>
          </material>
          <cast_shadows>1</cast_shadows>
          <transparency>0</transparency>
        </visual>
        <collision name="gate_3_collision_1">
          <pose frame="">-0.676776695297 0.676776695297 0.2 0 0 2.35619449019</pose>
          <geometry>
            <box>
              <size>0.1 0.02 0.4</size>
            </box>
          </geometry>
        </collision>
        <pose frame="">0 0 0 0 0 0</pose>
        <gravity>0</gravity>
        <self_collide>0</self_collide>
        <kinematic>1</kinematic>
        <visual name="gate_3_visual_2">
          <pose frame="">-0.5 0.5 1.2 0 0 2.35619449019</pose>
          <geometry>
            <box>
              <size>0.6 0.02 1.6</size>
            </box>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/DarkOrangeTransparentOverlay</name>
            </script>
          </material>
          <cast_shadows>1</cast_shadows>
          <transparency>0</transparency>
        </visual>
        <collision name="gate_3_collision_2">
          <pose frame="">-0.5 0.5 1.2 0 0 2.35619449019</pose>
          <geometry>
            <box>
              <size>0.6 0.02 1.6</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>
    <model name="gate_4">
      <pose frame="''">0 0 0 0 0 0</pose>
      <link name="link_0">
        <pose frame="">0 0 0 0 0 0</pose>
        <gravity>0</gravity>
        <self_collide>0</self_collide>
        <kinematic>1</kinematic>
        <visual name="gate_4_visual_0">
          <pose frame="">-0.676776695297 -0.676776695297 0.2 0 0 0.785398163397</pose>
          <geometry>
            <box>
              <size>0.1 0.02 0.4</size>
            </box>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/DarkOrangeTransparentOverlay</name>
            </script>
          </material>
          <cast_shadows>1</cast_shadows>
          <transparency>0</transparency>
        </visual>
        <collision name="gate_4_collision_0">
          <pose frame="">-0.676776695297 -0.676776695297 0.2 0 0 0.785398163397</pose>
          <geometry>
            <box>
              <size>0.1 0.02 0.4</size>
            </box>
          </geometry>
        </collision>
        <pose frame="">0 0 0 0 0 0</pose>
        <gravity>0</gravity>
        <self_collide>0</self_collide>
        <kinematic>1</kinematic>
        <visual name="gate_4_visual_1">
          <pose frame="">-0.323223304703 -0.323223304703 0.2 0 0 0.785398163397</pose>
          <geometry>
            <box>
              <size>0.1 0.02 0.4</size>
            </box>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/DarkOrangeTransparentOverlay</name>
            </script>
          </material>
          <cast_shadows>1</cast_shadows>
          <transparency>0</transparency>
        </visual>
        <collision name="gate_4_collision_1">
          <pose frame="">-0.323223304703 -0.323223304703 0.2 0 0 0.785398163397</pose>
          <geometry>
            <box>
              <size>0.1 0.02 0.4</size>
            </box>
          </geometry>
        </collision>
        <pose frame="">0 0 0 0 0 0</pose>
        <gravity>0</gravity>
        <self_collide>0</self_collide>
        <kinematic>1</kinematic>
        <visual name="gate_4_visual_2">
          <pose frame="">-0.5 -0.5 1.2 0 0 0.785398163397</pose>
          <geometry>
            <box>
              <size>0.6 0.02 1.6</size>
            </box>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/DarkOrangeTransparentOverlay</name>
            </script>
          </material>
          <cast_shadows>1</cast_shadows>
          <transparency>0</transparency>
        </visual>
        <collision name="gate_4_collision_2">
          <pose frame="">-0.5 -0.5 1.2 0 0 0.785398163397</pose>
          <geometry>
            <box>
              <size>0.6 0.02 1.6</size>
            </box>
          </geometry>
        </collision>
      </link>
    </model>
    <model name="sign_stop">
      <pose frame="''">2.0 3.0 0.0 -1.57079632679 0.0 1.57079632679</pose>
      <link name="link_0">
        <pose frame="''">0 0 0 0 0 0</pose>
        <visual name="visual">
          <geometry>
            <box>
              <size>0.2 0.2 1e-5</size>
            </box>
          </geometry>
          <material>
            <script>
              <uri>file:///home/dduberg/catkin_ws/src/dd2419/dd2419_simulation/signs/0.2x0.2/scripts</uri>
              <uri>file:///home/dduberg/catkin_ws/src/dd2419/dd2419_simulation/signs/0.2x0.2/textures</uri>
              <name>sign_stop/sign</name>
            </script>
          </material>
          <pose frame="''">0 0 0.001 0 0 0</pose>
          <cast_shadows>1</cast_shadows>
          <transparency>0</transparency>
        </visual>
        <collision name="collision">
          <pose frame="''">0 0 0.001 0 0 0</pose>
          <geometry>
            <box>
              <size>0.2 0.2 1e-5</size>
            </box>
          </geometry>
        </collision>
        <visual name="visual_backside">
          <pose frame="''">0 0 0 0 0 0</pose>
          <geometry>
            <box>
              <size>0.2 0.2 1e-5</size>
            </box>
          </geometry>
          <material>
            <script>Gazebo/Grey</script>
          </material>
        </visual>
        <collision name="collision_backside">
          <pose frame="''">0 0 0 0 0 0</pose>
          <geometry>
            <box>
              <size>0.2 0.2 1e-5</size>
            </box>
          </geometry>
        </collision>
      </link>
      <static>1</static>
      <allow_auto_disable>1</allow_auto_disable>
    </model>
    <physics default="0" name="default_physics" type="ode">
      <gravity>0 0 -9.8066</gravity>
      <ode>
        <solver>
          <type>quick</type>
          <iters>20</iters>
          <sor>1.3</sor>
          <use_dynamic_moi_rescaling>0</use_dynamic_moi_rescaling>
        </solver>
        <constraints>
          <cfm>0</cfm>
          <erp>0.2</erp>
          <contact_max_correcting_vel>100</contact_max_correcting_vel>
          <contact_surface_layer>0.001</contact_surface_layer>
        </constraints>
      </ode>
      <max_step_size>0.002</max_step_size>
      <real_time_factor>1</real_time_factor>
      <real_time_update_rate>500</real_time_update_rate>
      <magnetic_field>6e-06 2.3e-05 -4.2e-05</magnetic_field>
    </physics>
  </world>
</sdf>
